src/codec.rs

```diff
@@ -77,6 +81,7 @@ impl Codec {
             Self::Uper => crate::uper::decode(input),
             Self::Oer => crate::oer::decode(input),
             Self::Coer => crate::coer::decode(input),
+            Self::Xer => crate::coer::decode(input),
```

Comment:
Shouldn't this be xer?


```suggestion
            Self::Xer => crate::xer::decode(input),
```